pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go

```diff
@@ -927,6 +927,40 @@ func TestPickOneNodeForPreemption(t *testing.T) {
 	}
 }
 
+func TestPodEligibleToPreemptOthers(t *testing.T) {
+	tests := []struct {
+		name          string
+		pod           *v1.Pod
+		pods          []*v1.Pod
+		nodes         []string
+		nodesStatuses framework.NodeToStatusMap
+		expected      bool
+	}{
+		{
+			name:  "Pod with nominated node",
+			pod:   &v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "machine1", UID: types.UID("machine1")}, Spec: v1.PodSpec{Containers: smallContainers}, Status: v1.PodStatus{NominatedNodeName: "machine1"}},
+			pods:  []*v1.Pod{},
+			nodes: []string{"machine1"},
+			nodesStatuses: framework.NodeToStatusMap{
+				"machine1": framework.NewStatus(framework.UnschedulableAndUnresolvable, tainttoleration.ErrReasonNotMatch),
+			},
+			expected: true,
+		},
+	}
+
+	for _, test := range tests {
+		var nodes []*v1.Node
+		for _, n := range test.nodes {
+			nodes = append(nodes, makeNode(n, schedutil.DefaultMilliCPURequest*5, schedutil.DefaultMemoryRequest*5))
+		}
+		snapshot := internalcache.NewSnapshot(test.pods, nodes)
```

Comment:
You don't need to build a snapshot **_just for_** fetching nodeInfos. Just build the nodeInfos directly, something like:

```go
var nodeInfos []*framework.NodeInfo
for _, name := range nodeNames {
	ni := framework.NewNodeInfo()
	ni.SetNode(st.MakeNode().Name(name).Obj())
	nodeInfos = append(nodeInfos, ni)
}
```

Reply:
got it, thanks!

Reply:
sorry, I continue use the snapshot, since the podEligibleToPreemptOthers function is looking for 'framework.NodeInfoLister' which snapshot implement it